commit 86c3d4af6a7ed68e4166164a163586b906788a3a
Author: Alexander K <ak@tempesta-tech.com>
Date:   Mon Jan 28 02:17:30 2019 +0300

    Fix up TSO segments after TLS overhead; the previous warning may fail.

diff --git a/net/ipv4/tcp_output.c b/net/ipv4/tcp_output.c
index 14918c14..3066c3ac 100644
--- a/net/ipv4/tcp_output.c
+++ b/net/ipv4/tcp_output.c
@@ -2403,9 +2403,8 @@ static bool tcp_write_xmit(struct sock *sk, unsigned int mss_now, int nonagle,
 				tcp_reset(sk);
 				break;
 			}
-			/* We must not break TSO. */
-			WARN_ON_ONCE(tcp_skb_pcount(skb)
-				     != DIV_ROUND_UP(skb->len, mss_now));
+			/* Fix up TSO segments after TLS overhead. */
+			tcp_set_skb_tso_segs(skb, mss_now);
 		}
 #endif
 		if (unlikely(tcp_transmit_skb(sk, skb, 1, gfp)))
